packages/mesh/app/media/page.tsx

```diff
@@ -0,0 +1,99 @@
+import { getClient } from '@/apollo'
+import {
+  type GetPublishersQuery,
+  GetLatestStoriesDocument,
+  GetMostPickedStoryDocument,
+  GetPublishersDocument,
+  ListStoryFragment,
+} from '@/graphql/__generated__/graphql'
+
+import CategorySelector from './_components/category-selector'
+import Media from './_components/media'
+
+export const revalidate = 0
+
+type Story = ListStoryFragment
+type Publisher = NonNullable<GetPublishersQuery['publishers']>[number]
+
+export default async function Page() {
+  const mockMostPickedStoryId = '1175202'
+  const pageStoriesCount = 20
+  const mediaCount = 5
+  let stories: Story[] = []
+  let mostPickedStory: Story | null | undefined
+  let publishers: Publisher[] = []
+
+  const responses = await Promise.allSettled([
+    fetchLatestStories({ take: pageStoriesCount }),
+    fetchMostPickedStory({ id: mockMostPickedStoryId }),
+    fetchMedia({ take: mediaCount }),
+  ])
+  if (responses[0].status === 'fulfilled') {
+    stories = responses[0].value.data.stories ?? []
+  } else {
+    // TODO: handle gql failed error
+    console.error(responses[0].reason)
+  }
+
+  if (responses[1].status === 'fulfilled') {
+    mostPickedStory = responses[1].value.data.story
+  } else {
+    // TODO: handle gql failed error
+    console.error(responses[1].reason)
+  }
+
+  if (responses[2].status === 'fulfilled') {
+    publishers = responses[2].value.data.publishers ?? []
+  } else {
+    // TODO: handle gql failed error
+    console.error(responses[2].reason)
+  }
+
+  // TODO: fetch real publiser stories
+  const displayPublishers = publishers.map((publisher) => ({
+    ...publisher,
+    stories: stories?.slice(0, 3) ?? [],
+  }))
+
+  return (
+    <main className="bg-white">
```

Comment:
<img width="765" alt="Screenshot 2024-06-16 at 3 32 08 PM" src="https://github.com/readr-media/Sachiel/assets/114332515/42b41d9d-0394-4eed-8b8b-6f550901e391">

1920px, 右側不是 white

Reply:
修改在 [48a18f7](https://github.com/readr-media/Sachiel/pull/774/commits/48a18f765742d2bf68b653441dc78ba75275b176)

Reply:
我個人比較傾向 `media/layout.tsx`
```javascript
export default function MediaLayout({
  children,
}: {
  children: React.ReactNode
}) {
  return (
    <div className="bg-white">
      <div className="grow xl:max-w-[theme(width.maxMain)]">{children}</div>
    </div>
  )
}

```

```javascript

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode
}>) {
  return (
    <html lang="zh-Hant" className={notoSans.className}>
      <body className="min-h-screen">
        {/* fixed header */}
        <Header />
        {/* block for non-fixed content, set padding for fixed blocks */}
        <div className="flex min-h-screen flex-col pb-[theme(height.nav.default)] pt-[theme(height.header.default)]  sm:pb-0 sm:pt-[theme(height.header.sm)] sm:pl-[theme(width.nav.sm)] md:pl-[theme(width.nav.md)] xl:pl-[calc((100vw-theme(width.maxContent))/2+theme(width.nav.xl))]">
          {/* block for main and aside content to maintain the max width for screen width larger than 1440 */}
          {children}
          {/* footer after main content */}
          <Footer />
        </div>
        {/* fixed nav, mobile on the bottom, otherwise on the left side */}
        <Nav />
      </body>
    </html>
  )
}

```

Reply:
目前設計稿看已完工的頁面其實還是會有白色和灰色兩種版本，目前的設計是把 `<div className="grow xl:max-w-[theme(width.maxMain)]">{children}</div>` 的寬度限制保留在 layout 中，畢竟全部吃這個 layout 的頁面都會是同一個邏輯，放到 media layout 反而會變成任何新的 page 都會需要加上類似 media layout  的寫法 (包含 social page)，我認為沒有比較方便

Reply:
https://nextjs.org/docs/app/building-your-application/routing/route-groups
你們可能需要用 router group 來管理右邊背景色不同的情況，反正現在頁面還不多的情況下，就先處理吧，這可以另外開 PR 處理這部份的問題

Reply:
目前想法是 router group 會用在一般頁面和登入頁面來區別不同 layout，背景色不同要不要用這個方式管理目前是覺得不用特別加 router group 來管理 (怕太多層 router group)，不過之後頁面多了還是可以討論再做更好的處理

Reply:
但目前解法應該是在 client-side 做切換？

Reply:
對啊，因為 layout 裡面沒有辦法直接拿到 path 來做判斷，不過目前的寫法只有 Content 會是 client component，page.tsx 以下都還是 server component，好奇這樣有什麼問題嗎？